servers/src/websocket/farmer/handshake.rs

```diff
@@ -29,23 +30,34 @@ impl MessageHandler for HandshakeHandle {
         peers: PeerMap,
     ) -> Result<(), Error> {
         let mut cursor = Cursor::new(&msg.data);
-        let handshake = Handshake::from_bytes(&mut cursor)?;
+        let peer = peers.read().await.get(&peer_id).cloned();
+        let protocol_version = if let Some(peer) = peer.as_ref() {
+            *peer.protocol_version.read().await
+        } else {
+            ChiaProtocolVersion::default()
+        };
+        let handshake = Handshake::from_bytes(&mut cursor, protocol_version)?;
         debug!("New Peer: {}", &peer_id);
-        if let Some(peer) = peers.lock().await.get(&peer_id).cloned() {
+        if let Some(peer) = peers.read().await.get(&peer_id).cloned() {
             let (network_id, server_port) = {
                 let cfg = self.config.clone();
                 (cfg.network.clone(), cfg.websocket.port)
             };
-            *peer.node_type.lock().await = NodeType::from(handshake.node_type);
+            *peer.node_type.write().await = NodeType::from(handshake.node_type);
+            let protocol_version = ChiaProtocolVersion::from_str(&handshake.protocol_version)
+                .expect("ChiaProtocolVersion::from_str is Infallible");
+            *peer.protocol_version.write().await = protocol_version;
             peer.websocket
                 .lock()
                 .await
                 .send(Message::Binary(
                     ChiaMessage::new(
                         ProtocolMessageTypes::Handshake,
+                        protocol_version,
                         &Handshake {
                             network_id,
-                            protocol_version: PROTOCOL_VERSION.to_string(),
+                            //Server Will use version sent by peer
+                            protocol_version: protocol_version.to_string(),
```

Comment:
We should probably only send the peers version if we support it, that being every version below the highest currently implemented one (assuming we support all old versions), but not a version higher

For example:
- peer with 0.0.34 connects, we send back 0.0.34, as we support it
- peer with 0.0.42 connects, we send back 0.0.36, as that's the highest we support (right now)

That way it is clear to the client whether we actually support the version we send, or not

Reply:
The from_str call above will detect known versions, (0.0.34-36) or will fall back to the default. So this behavior should already be in place.